firebase-ml-modeldownloader/src/main/java/com/google/firebase/ml/modeldownloader/internal/CustomModelDownloadService.java

```diff
@@ -147,17 +147,18 @@ public CustomModelDownloadService(
             if (!installationAuthTokenTask.isSuccessful()) {
               ErrorCode errorCode = ErrorCode.MODEL_INFO_DOWNLOAD_CONNECTION_FAILED;
               String errorMessage = "Failed to get model due to authentication error";
+              int exceptionCode = FirebaseMlException.UNAUTHENTICATED;
               if (installationAuthTokenTask.getException() != null
                   && (installationAuthTokenTask.getException() instanceof UnknownHostException
                       || installationAuthTokenTask.getException().getCause()
                           instanceof UnknownHostException)) {
                 errorCode = ErrorCode.NO_NETWORK_CONNECTION;
                 errorMessage = "Failed to retrieve model info due to no internet connection.";
+                exceptionCode = FirebaseMlException.INTERNAL;
```

Comment:
I wonder if we should surface network errors... I mean, no network connection is something the developer might want to surface to the end-user. INTERNAL errors should be limited to something that's our fault...